src/backend/gporca/libgpopt/src/xforms/CXformUtils.cpp

```diff
@@ -1340,7 +1338,7 @@ CXformUtils::PexprLogicalDMLOverProject(CMemoryPool *mp,
 	IMDId *rel_mdid = ptabdesc->MDId();
 	CExpression *pexprProject = NULL;
 	CColRef *pcrAction = NULL;
-	CColRef *pcrOid = NULL;
+	CColRef *pcrOid = pcrTableOid;
```

Comment:
In this function we have to fully remove the logic related with definition of partition oid column as we define this column from outside

Reply:
Removing logic of generation pcrOid for Insert operation (and using the pcrOid from the function argument (in case of insert it's always `NULL`)) leads to `SIGSEGV` at next test cases of `CDMLTest`.

- src/backend/gporca/data/dxl/minidump/Insert-Parquet-Partitioned.mdp
- src/backend/gporca/data/dxl/minidump/Insert-AO-Partitioned.mdp

For now pcrOid is `NULL` and the problem occurs at this [code fragment (NULL pointer dereference)](https://github.com/arenadata/gpdb/blob/677a3cbda8b571709a35c5341d0024f9f38a63a6/src/backend/gporca/libgpopt/src/operators/CPhysicalDML.cpp#L531-L537).

For the first test case `pcrOid` was generated, because it's a `parquet` table ([`FGeneratePartOid` returns true]((https://github.com/arenadata/gpdb/blob/677a3cbda8b571709a35c5341d0024f9f38a63a6/src/backend/gporca/libgpopt/src/base/CUtils.cpp#L4410-L4429))
For the second test case [Hint option is used](https://github.com/arenadata/gpdb/blob/677a3cbda8b571709a35c5341d0024f9f38a63a6/data/dxl/minidump/Insert-AO-Partitioned.mdp#L14) while at adb-6.x [COptTask sets this hint to int_max](https://github.com/arenadata/gpdb/blob/677a3cbda8b571709a35c5341d0024f9f38a63a6/src/backend/gpopt/utils/COptTasks.cpp#L397).

So logic was changed - pcrOid is generated only for insert operation